### PR TITLE
GH-15857: AutoML pipeline support

### PR DESCRIPTION
**Pull request by sebhrusen**
integrate TE in AutoML as a pipeline transformer: https://github.com/h2oai/h2o-3/issues/15857

### CODE REVIEW

File: h2o-automl/src/main/java/ai/h2o/automl/AutoML.java [hunk 1]
```diff
@@ -387,11 +392,53 @@ private void initLeaderboard() {
...
+    // and multiple transformers can refer 
+      // to the same model, 
+    // then we should be careful when deleting a transformer (resp. an entire pipeline) 
+    // as we may delete sth that is still in use by another transformer (resp. pipeline).
+    // --> ref count?
+    
+    //TODO: in AutoML, the same transformations are likely to occur on multiple (sometimes all) models, 
+    // especially if the transformers parameters are not tuned.
+    // But it also depends if the transformers are context(CV)-sensitive (e.g. Target Encoding).
+    // See `CachingTransformer` for some thoughts about this.
```
Comment (sebhrusen) [inline h2o-automl/src/main/java/ai/h2o/automl/AutoML.java:426]: these are just things to keep in mind, but not currently needed

**Review by wendycwong — APPROVED** [review]

File: h2o-automl/src/main/java/water/automl/api/schemas3/AutoMLBuildSpecV99.java [hunk 2]
```diff
@@ -293,6 +293,10 @@ public static final class AutoMLBuildModelsV99 extends SchemaV3<AutoMLBuildSpec.
     @API(help = "A mapping representing monotonic constraints. Use +1 to enforce an increasing constraint and -1 to specify a decreasing constraint.",
             level = API.Level.secondary)
     public KeyValueV3[] monotone_constraints;
+    
+    @API(help = "Temporary flag to disable pipeline usage when preprocessing is enabled.", 
+            level = API.Level.expert)
+    public boolean pipelineEnabled;
```
Comment (tomasfryda) [inline h2o-automl/src/main/java/water/automl/api/schemas3/AutoMLBuildSpecV99.java:299]: This seems to be impossible to use using the python API. Is it intentional?

**Review by tomasfryda — APPROVED** [review]
Looks good to me, thank you @sebhrusen !

File: h2o-automl/src/main/java/water/automl/api/schemas3/AutoMLBuildSpecV99.java (hunk 2 above)
Comment (sebhrusen) [reply h2o-automl/src/main/java/water/automl/api/schemas3/AutoMLBuildSpecV99.java:299]: good catch, but as I'm removing the `AutoMLBuildSpec.pipelineEnabled` property in the cleanup PR (https://github.com/h2oai/h2o-3/pull/16061/) and I also forgot to remove it from the schema there, I'll do the change in that last PR if you don't mind.

Comment (tomasfryda) [reply h2o-automl/src/main/java/water/automl/api/schemas3/AutoMLBuildSpecV99.java:299]: yes, that's fine.

**Comment by sebhrusen** [comment]
@wendycwong, @tomasfryda this one is ready for merge.
Double checked the failing AutoML test (http://mr-0xg1:8080/job/h2o-3-pipeline/job/PR-16041/12/testReport/junit/ai.h2o.automl/AutoMLTest/Java_8_AutoML_JUnit___test_algos_have_default_parameters_enforcing_reproducibility/), ran in multiple times locally (single+multi nodes) without key leakage so don't know what happened there: looks like the test got interrupted.

**Review by tomasfryda — APPROVED** [review]

**Pull request merged**